test/topology_experimental_raft/test_tablets.py

```diff
@@ -464,6 +464,44 @@ async def check_repair_history():
 
     await cql.run_async("DROP KEYSPACE test;")
 
+@pytest.mark.repair
+@pytest.mark.asyncio
+async def test_tablet_repair_ranges_selection(manager: ManagerClient):
+    logger.info("Bootstrapping cluster")
+    servers = [await manager.server_add(), await manager.server_add()]
+
+    rf = 2
+    tablets = 4
+
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE test WITH replication = {{'class': 'NetworkTopologyStrategy', "
+                  "'replication_factor': {}}} AND tablets = {{'initial': {}}};".format(rf, tablets))
+    await cql.run_async("CREATE TABLE test.test (pk int PRIMARY KEY, c int) WITH tombstone_gc = {'mode':'repair'};")
+
+    logger.info("Populating table")
+
+    keys = range(256)
+    await asyncio.gather(*[cql.run_async(f"INSERT INTO test.test (pk, c) VALUES ({k}, {k});") for k in keys])
+
+    hosts = await wait_for_cql_and_get_hosts(cql, servers, time.time() + 60)
+    logging.info(f'Got hosts={hosts}');
+
+    await repair_on_node(manager, servers[0], servers, ranges='-4611686018427387905:-1,4611686018427387903:9223372036854775807')
```

Comment:
shouldn't tablet load-balancing be disabled?

Reply:
We are not going to disable load balancing when user runs repair. Isn't it better to test how people actually runs it.  And would the load balancing be triggered at all when there are such few data?

Reply:
How can we not disable load balancing during repair until we provide a cluster-repair api that will schedule tablet repair natively and in synchronization with the tablet scheduler?

Reply:
It should be part of 1) above: https://github.com/scylladb/scylladb/pull/17436#discussion_r1503531819

Reply:
Yes, I had concerns about it not being reenabled, but we decided to go through with this approach.
I disagree that's in not a big deal to silently not repair some tablets since they were migrated concurrently to async repair (or sue to split/merge).  Repair is not supposed to be a "best effort" procedure, and retrying at a later time does not ensure you that the tablet won't be migrated yet again.
With the existing apis one need to freeze the tablet mapping for taking a snapshot and during repair.
If this is a no go, then we must implement native, cluster-wide tablet repair, yet I think it's a reasonable constraint for first release.
Cc @tzach @dani-tweig

Reply:
Scylla Manger supports task pausing/resuming.
When repair of vnode table is resumed, SM queries ring description and just skips all token ranges that have been successfully repaired in the previous run (stored in SM DB).

What about resuming repair of tablet table? Is it safe to skip  successful ranges as in case of vnode tables (not sure about that)?

> It is not a big deal if the scheduled range is not repaired due to the migration, it could be repaired by the next scheduled repair.

Maybe it's true with a constraint that all tablet tables have `tombstone_gc = {'mode': 'repair'}`?

Reply:
> Yes, I had concerns about it not being reenabled, but we decided to go through with this approach.
I disagree that's in not a big deal to silently not repair some tablets since they were migrated concurrently to async repair (or sue to split/merge). Repair is not supposed to be a "best effort" procedure, and retrying at a later time does not ensure you that the tablet won't be migrated yet again.
With the existing apis one need to freeze the tablet mapping for taking a snapshot and during repair.
If this is a no go, then we must implement native, cluster-wide tablet repair, yet I think it's a reasonable constraint for first release.

Yes, we plan to require tombstone_gc = repair for tablets tables. That's why it is safe to skip the ranges that is scheduled to a wrong node due to the migration. There is no safety issue.